app/services/invoices/create_service.rb

```diff
@@ -82,6 +82,10 @@ def create_charges_fees(invoice, subscription, boundaries)
     end
 
     def should_create_subscription_fee?(subscription)
+      # NOTE: When plan is pay in advance we generate an invoice upon subscription creation
+      # We want to prevent creating subscription fee if subscription creation already happened on billing day
+      return false if subscription.plan.pay_in_advance? && subscription.already_billed_today?
```

Comment:
Shouldn't we check for the day in the "timestamp" rather than today?

Reply:
oh yes, you are right, thanks for catching! Fixed